### ISSUE TITLE
[Feature] Add provisioner for DIY VPS

### ISSUE THREAD

**Issue by e-minguez** [inlets/inlets-operator#27]
## Expected Behaviour
[Feature] Add provisioner for VPS

## Current Behaviour
Exit nodes can be provisioned to DO/Packet

## Possible Solution
Instead deploying a new instance, use an already created VPS/custom VM instead.

## Steps to Reproduce (for bugs)
1. Deploy the inlets operator
2. Tweak the config with a configmap
3. Add the token with a secret
4. Profit

## Context
Inlets does support DIY VPS/VMs

**Comment by alexellis** [comment]
Great to hear that you're interested in inlets. I'm not sure what you mean by "DIY VPS", could you elaborate on what you're proposing? Thanks

**Comment by e-minguez** [comment]
> Great to hear that you're interested in inlets. I'm not sure what you mean by "DIY VPS", could you elaborate on what you're proposing? Thanks

So, inlets does support to be deployed in a DIY VPS (https://github.com/inlets/inlets/blob/master/docs/vps.md), meaning you can create your own VM in the provider you want (or baremetal, or $whatever), deploy inlets and use it.

The current inlets-operator only supports DO, Packet.net and other providers (which is fine), but it would be nice to allow the inlets-operator to use an already created inlets deployment instead deploying your own instance $somewhere.

Is that clear enough? Thanks!!!

**Comment by alexellis** [comment]
No, it's still not very clear.

Can you list the workflow you imagine being available after this work would be completed?

Are you proposing to do it yourself?

The current list of providers and incoming providers are listed on the README.md. 

Alex

**Comment by e-minguez** [comment]
> No, it's still not very clear.
> 

Let me try again.

> Can you list the workflow you imagine being available after this work would be completed?
> 

Environment:
* myserver.example.com: Linux OS with inlets installed and listening (`inlets server --port 8000 --token=asdf`)
* k8s deployment.

Use case:
As a user, I want to be able to use the inlets-operator with myserver.exampl.ecom inlets installation so I don't need to provision anything in any cloud/provider.

Procedure:

```
kubectl create secret generic inlets-token \
  --from-literal inlets-token="asdf"
kubectl apply -f ./artifacts/crd.yaml

kubectl apply -f ./artifacts/operator-rbac.yaml
kubectl apply -f ./artifacts/operator.yaml
```

I'm assuming some ConfigMap or an env var would be required to specify the server url (myserver.example.com)

> Are you proposing to do it yourself?
> 

It would be nice, but unfortunately I don't have the skills nor the time to do it.

> The current list of providers and incoming providers are listed on the README.md.
> 

:+1: 

> Alex

**Comment by alexellis** [comment]
I think you could just create the Tunnel custom resource yourself with YAML. Would you like to try that?

**Comment by e-minguez** [comment]
> I think you could just create the Tunnel custom resource yourself with YAML. Would you like to try that?

You mean instead deploying the operator, just run inlets in k8s right? Do you have any example that I can look at?

Thanks.

**Comment by alexellis** [comment]
Feel free to experiment with inletsctl or the tunnel CRD. I'm going to close the issue since there isn't a clear set of requirements I can follow at this stage. 

For technical support, please feel free to use Slack
